internal/service/resources/resources.go

```diff
@@ -0,0 +1,63 @@
+package resources
+
+import (
+	"context"
+	"github.com/go-kratos/kratos/v2/log"
+	authnapi "github.com/project-kessel/inventory-api/internal/authn/api"
+	"github.com/project-kessel/inventory-api/internal/biz/model"
+	"github.com/project-kessel/inventory-api/internal/biz/resources"
+	"github.com/project-kessel/inventory-api/internal/middleware"
+	conv "github.com/project-kessel/inventory-api/internal/service/common"
+
+	pb "github.com/project-kessel/inventory-api/api/kessel/inventory/v1beta2"
+)
+
+type ResourceService struct {
+	pb.UnimplementedKesselResourceServiceServer
+	Ctl *resources.Usecase
+}
+
+func New(c *resources.Usecase) *ResourceService {
+	return &ResourceService{
+		Ctl: c,
+	}
+}
+
+func (c *ResourceService) ReportResource(ctx context.Context, r *pb.ReportResourceRequest) (*pb.ReportResourceResponse, error) {
+	identity, err := middleware.GetIdentity(ctx)
+	if err != nil {
+		return nil, err
+	}
+	if h, err := toResource(r, identity); err == nil {
+		if resp, err := c.Ctl.Create(ctx, h); err == nil {
+			return fromResource(resp), nil
+
+		} else {
+			return nil, err
+		}
+	} else {
+		return nil, err
+	}
```

Comment:
nit: this largely comes down to preference but go teams I have been on have preferred not to use inline error statements to make things more readable, that would look something like...

```
h, err := toResource(r, identity)
if err != nil {
	return nil, err
}

resp, err := c.Ctl.Create(ctx, h)
if err != nil {
	return nil, err
}

return fromResource(resp), nil
```

Reply:
I like this as well, compared to a 100 nested if blocks. Update made.